20200314_module_stage_task

aristasoluciones opened the pull request:
se agrega paginacion en modulo stage

Commit:
Merge branch 'master' into 20200226_module_stage_task
20200304 merge whit master branch

hbkruzpe committed:
paginacion en stage y task

hbkruzpe committed:
clean directory

Commit:
Merge branch 'master' into 20200226_module_stage_task

mwleinad commented on Modules/Stage/Models/Stage.php:
```diff
@@ -5,10 +5,15 @@
 use Illuminate\Database\Eloquent\Model;
 use Illuminate\Database\Eloquent\SoftDeletes;
 use Modules\Task\Models\Task;
+use \Askedio\SoftCascade\Traits\SoftCascadeTrait;
```
se te paso el \ al principio

mwleinad commented on Modules/Stage/Services/StagePatchService.php:
```diff
@@ -22,19 +24,22 @@ class StagePatchService {
+     * @param array $data
+     * @return array
+     */
+    private function normalizeData(User $user, array $data) {
```
digo, en teoria esto no seria necesario no? 
mi recomendacion es solo poner esta funcion si se requiere hacer algun tipo de normalizacion ya que ahorita solo estas regresando lo mismo

mwleinad commented on Modules/Task/Http/Requests/TaskGetValidationRequest.php:
```diff
@@ -11,24 +11,33 @@
-        return true;
+    public function rules() {
+        return [
+            'predicates' => 'array|nullable',
```
Esto no es necesario hacerlo pero si nos fijamos este codigo esta repetido varias veces, se puede hacer 2 cosas
un validador customizado para poner algo como 
'predicates => 'array|nullable|predicates' (esto manda a llamar a una funcion personalizada)
o
en el config de validacion agregas todo este arreglo desde predicates hasta per_page y haces algo como
return config('validation.predicates')

mwleinad commented on composer.json:
```diff
@@ -9,13 +9,14 @@
         "nwidart/laravel-modules": "^6.2",
-        "tymon/jwt-auth": "1.0.0-rc.5.1",
-        "ext-json": "*"
+        "tymon/jwt-auth": "1.0.0-rc.5.1"
```
porque le quitaste la de ext-json?

aristasoluciones replied:
No la quite, se acomodo mas arriba, saber que hizo composer

hbkruzpe committed:
MOD: quitar validation predicates al listar tareas.
MOd: reglas de validacion para metodo GET se coloca en un archivo de configuracion

aristasoluciones reviewed (commented):
Me lo revisas

hbkruzpe committed:
Merge branch 'master' into 20200226_module_stage_task

# Conflicts:
#	modules_statuses.json

mwleinad commented on config/validation.php:
```diff
@@ -0,0 +1,24 @@
+    | the next methods of http [GET]
+    |
+    */
+    'default' => [
```
yo lo llamaria default_http_get 
para que sea mas "verboso"

hbkruzpe committed:
rename default to default_http_get into config/validation.php
